src/main/java/cz/sparko/boxitory/service/filesystem/FilesystemHashStore.java

```diff
@@ -0,0 +1,131 @@
+        }
+
+        try {
+            String boxHashFilename = box + algorithm.getFileExtension();
+            File hashFile = new File(boxHashFilename);
+            if (hashFile.exists()) {
+                LOG.trace("Hash file [{}] already exist. Not replacing!", boxHashFilename);
+                return;
+            }
+            boolean fileCreated = hashFile.createNewFile();
+            if (fileCreated) {
+                try (FileWriter writer = new FileWriter(hashFile)) {
+                    writer.write(createValidChecksumFilecontent(hash, boxFile));
+                    LOG.debug("Storing hash file [{}]", hashFile.getName());
+                }
+            } else {
+                LOG.debug("Hash file [{}] was not created. Dropping.");
+            }
+        } catch (IOException e) {
+            LOG.error("Something went wrong when persisting hash file.", e);
+        }
+    }
+
+    /**
+     * Reads checksum from filesystem, from file beside box file with extension derived from {@link HashAlgorithm}.
+     *
+     * @param box       absolute filesystem path to the box
+     * @param algorithm algorithm of hash
+     * @return hash when file found, {@link Optional#empty()} otherwise
+     * @throws IllegalStateException when file provided by {@code box} does not exists
+     */
+    @Override
+    public Optional<String> loadHash(String box, HashAlgorithm algorithm) {
+        File boxFile = checkBoxFileExists(box);
+
+        final Optional<String> hash;
+
+        File boxHashFile = new File(box + algorithm.getFileExtension());
+        if (boxHashFile.exists() && boxHashFile.isFile()) {
+            LOG.trace("Found hash file [{}] for box [{}]", boxHashFile.getAbsolutePath(), box);
+            try {
+                hash = Optional.of(readHashFromChecksumFile(boxHashFile, boxFile.getName()));
+                LOG.trace("Hash [{}] loaded from file [{}] for box [{}]", hash, boxHashFile.getAbsolutePath(), box);
+            } catch (IllegalStateException ise) {
+                LOG.error("Checksum file [{}] has probably wrong format.", boxHashFile, ise);
+                return Optional.empty();
+            } catch (IOException e) {
+                LOG.error("Something went wrong when reading hash file.", e);
+                return Optional.empty();
+            }
+        } else {
+            hash = Optional.empty();
+        }
+        return hash;
+    }
+
+    /**
+     * Throw {@link IllegalStateException} when box file does not exists.
+     *
+     * @param box path to the box file
+     */
+    private File checkBoxFileExists(String box) {
+        File boxFile = new File(box);
+
+        if (!boxFile.exists() || !boxFile.isFile()) {
+            throw new IllegalStateException("box [" + box + "] does not exist!");
+        }
+
+        return boxFile;
+    }
+
+    private final String CHECKSUM_FILE_SEPARATOR = "  ";
+
+    private String createValidChecksumFilecontent(String hash, File forFile) {
```

Comment:
I see that there is javadoc at certain methods. I would keep consistence and add it to all methods.

Reply:
refactored and documented :)